enumerables.rb

```diff
@@ -0,0 +1,293 @@
+# rubocop:disable Metrics/CyclomaticComplexity, Metrics/MethodLength, Metrics/PerceivedComplexity, Metrics/AbcSize, Style/SymbolProc, Lint/MissingCopEnableDirective, Lint/RedundantCopDisableDirective
+module Enumerable
+  # 1.- ---- my_each Method ----
+
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    arr = to_a
+    arr.length.times do |indx|
+      yield(arr[indx])
+    end
+    self
+  end
+
+  # 2.- ---- my_each_with_index Method ----
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    target_arr = to_a
+    top = target_arr.length
+    0.upto(top - 1) do |i|
+      yield(target_arr[i], i)
+    end
+    self
+  end
+
+  # 3.- ---- my_select Method ----
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    if is_a? Hash
+      new_hsh = {}
+      my_each do |key, value|
+        new_hsh[key] = value if yield(key, value)
+      end
+      new_hsh
+    elsif is_a? String
+      'Wrong input'
+    else
+      new_arr = []
+      my_each do |elmt|
+        filtered_arr.push(elmt) if yield(elmt)
+      end
+      new_arr
+    end
+  end
+end
```

Comment:
* [ ] should have working `my_select` identical and returns the same thing as ruby's [`select`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-select). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_select.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

#### my_select
- [ ] my_select method has two problems. Please have a look below. 

- enumerables #**my_select** returns an array containing all elements of enum for which the given block returns a true value
<img width="659" alt="2" src="https://user-images.githubusercontent.com/58504655/98772744-41ab8f00-2409-11eb-8bf5-cfd85cb732c2.PNG">


- enumerable #my_select does not mutate the original array

<img width="503" alt="1" src="https://user-images.githubusercontent.com/58504655/98772707-28a2de00-2409-11eb-8f6b-803d177ca72c.PNG">

Reply:
Good evening Arslan Bisharat! Thanks for the review, I find your words very encouraging and the points you marked, done in time and focused right on the thing to fix! An example is this particular point. Indeed, as you show above, the reason that #my_select wasn't mutating the original array was that I used a variable with a different name than the one defined for this purpose. I've just changed it for the right one. Now, in the example below it really seems that is doing something to the given array:
(comparing #my_select with Ruby's #select) ![image](https://user-images.githubusercontent.com/63531796/98830387-fb056700-23ff-11eb-808f-11991d9c222f.png)
(now both outputs) ![image](https://user-images.githubusercontent.com/63531796/98830636-4455b680-2400-11eb-92c1-eea3fcff5558.png)